### PR TITLE
Fix the expansion error issue

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#5169

### CODE REVIEW

File: pkg/controller/control.go [hunk 1]
```diff
@@ -250,7 +253,7 @@ func (c *Controller) Expand(size int64) error {
 
 		expansionSuccess, errsNeedToBeHandled, errsForRecording := c.backend.Expand(size)
 		if errsForRecording != nil {
-			c.lastExpansionFailedAt = time.Now().UTC().Format(time.RFC3339)
+			c.lastExpansionFailedAt = time.Now().UTC().Format(time.RFC3339Nano)
```
Comment (innobead) [inline pkg/controller/control.go:256]: Why nano? for consistency?

**Review by innobead — CHANGES_REQUESTED** [review]
Don't we still need to do `c.lastExpansionFailedAt = ""` when the expansion succeed?

File: pkg/controller/control.go (hunk 1 above)
Comment (shuo-wu) [reply pkg/controller/control.go:256]: In the manager part, we will use `time.Now()` to do the comparison. And `time.Now()` is using nanosecond precision.

**Comment by shuo-wu** [comment]
> Don't we still need to do c.lastExpansionFailedAt = "" when the expansion succeed?

Just for recording, so that the manager will print out an event that explains why the replica becomes failed.

File: pkg/controller/control.go (hunk 1 above)
Comment (innobead) [reply pkg/controller/control.go:256]: I feel we should have a common library to use in every module, so we don't need to remember what time unit we are using.

**Comment by shuo-wu** [comment]
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2819/

File: pkg/controller/control.go (hunk 1 above)
Comment (shuo-wu) [reply pkg/controller/control.go:256]: YES. I was thinking if we need a util repo that covers all kinds of common functions. For example:
1. Rich error marshaling and unmarshaling
2. Address handling. (prefix, format, etc)
3. Timestamp handling.
4. Checksum handling.
5. UUID
......

Comment (shuo-wu) [reply pkg/controller/control.go:256]: YES. I was thinking if we need a util repo that covers all kinds of common functions. For example:
1. Rich error marshaling and unmarshaling
2. Address handling. (prefix, format, etc)
3. Timestamp handling.
4. Checksum handling.
5. UUID
......

Comment (shuo-wu) [reply pkg/controller/control.go:256]: YES. I was thinking if we need a util repo that covers all kinds of common functions. For example:
1. Rich error marshaling and unmarshaling
2. Address handling. (prefix, format, etc)
3. Timestamp handling.
4. Checksum handling.
5. UUID
......

Comment (innobead) [reply pkg/controller/control.go:256]: then, let's create it. https://github.com/longhorn/common-libs

**Comment by innobead** [comment]
@Mergifyio backport v1.4.x

**Pull request merged**